Orange/data/table.py

```diff
@@ -912,15 +912,16 @@ def is_copy(self):
 
     def ensure_copy(self):
         """
-        Ensure that the table owns its data; copy arrays when necessary
+        Ensure that the table owns its data; copy arrays when necessary.
+        The check is skipped for sparse matrices since they don't have views like numpy arrays.
         """
-        if self.X.base is not None:
+        if not sp.issparse(self.X) and self.X.base is not None:
```

Comment:
What about introducing a functions like

def is_view(x):
    return not sp.issparse(x) and x.base is not None

and using it in ifs?